app/modules/cron.c

```diff
@@ -86,12 +85,21 @@ static int lcron_parsedesc(lua_State *L, char *str, struct cronent_desc *desc) {
 static int lcron_create(lua_State *L) {
   // Check arguments
   char *strdesc = (char*)luaL_checkstring(L, 1);
-  void *newlist;
   luaL_checktype(L, 2, LUA_TFUNCTION);
-  // Parse description
-  struct cronent_desc desc;
-  lcron_parsedesc(L, strdesc, &desc);
-  // Allocate userdata
+
+  // Grab the table of all entries onto the stack
+  lua_rawgeti(L, LUA_REGISTRYINDEX, cronent_table_ref);
```

Comment:
`cronent_table_ref``contains a list of ud's and we scan for empty slots, scan for a specific one etc.  I just wonder if we would we better off using a keyed table with the ud as the key, and just let the table access functions do all this for us.

The comment isn't what the scan does, which is scan from the start of the table looking for the first null slot.

On a separate note, just in terms of efficiency scanning up a list for a null slot must terminate (possibly by throwing and EOM) so this scan could be more efficiently written:
```C
  while(lua_rawgeti(L, -1, ix) != LUA_TNIL) { lua_pop(L, 1); ix++; }
```

Reply:
Whoop, yes, the comment's stale.

I thought about using the `ud` as both the key and value, but this was a straightforward bring-over of existing logic, AIUI array-like tables are smaller in memory, and I think we should drop table entirely and just equip each cron entry with a timer as part of fixing #3160.